iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs

```diff
@@ -256,77 +261,135 @@ public override async Task SendEventAsync(IEnumerable<Message> messages, Cancell
 
         public override async Task<Message> ReceiveAsync(CancellationToken cancellationToken)
         {
-            cancellationToken.ThrowIfCancellationRequested();
+            if (_deviceReceiveMessageCallbackSet)
```

Comment:
In case we have set a callback for receiving messages, the call to ReceiveAsync() will always return null.